python/mxnet/gluon/rnn/rnn_layer.py

```diff
@@ -214,7 +215,7 @@ def begin_state(self, batch_size=0, func=ndarray.zeros, **kwargs):
             states.append(func(name='%sh0_%d'%(self.prefix, i), **info))
         return states
 
-    def hybrid_forward(self, F, inputs, states=None, **kwargs):
+    def hybrid_forward(self, F, inputs, sequence_length=None, states=None, **kwargs):
```

Comment:
for compatibility for positional argument, maybe move the new argument after states?

Reply:
I think `sequence_length` actually has to come before `states`, because it is a positional argument provided by the user, whereas `states` is an automatic positional argument provided by gluon, since gluon automatically adds a block's learnable parameters as positional arguments at the end of  `hybrid_forward()` functions' argument list.

The reason my personal unit tests were passing was because I was using a version of my code that hadn't switched the order of these arguments. I reverted this change so hopefully the centsos-gpu tests will pass now.

Reply:
Nevermind I misunderstood earlier.

Actually the `states` array is *possibly* a user-provided parameter, but if they don't provide it then the code will fill in a default value.

I was having trouble getting this to work:
```
model(input, sequence_length=sequence_length)
```
Because the code kept on complaining about an unknown parameter `sequence_length`. I think this must be related to how gluon handles keyword arguments for a model's forward pass.

The solution I found was to over-ride the `__call__()` method for the rnn layer which is happy to take keyword arguments, and then I can use that to impose an order for arguments that I pass to the `hybrid_forward()` function. To get this to work I also needed to move the logic of setting a default value for `states` into the `__call__()` function.

Reply:
right, hybrid_forward doesn't support keyword arguments.